OpenRA.Game/Actor.cs

```diff
@@ -96,7 +87,14 @@ internal Actor(World world, string name, TypeDictionary initDict)
 
 				Info = world.Map.Rules.Actors[name];
 				foreach (var trait in Info.TraitsInConstructOrder())
+				{
 					AddTrait(trait.Create(init));
+
+					// Some traits rely on properties provided by IOccupySpace in their initialization,
+					// so we must ready it now, we cannot wait until all traits have finished construction.
+					if (trait is IOccupySpaceInfo)
+						OccupiesSpace = Trait<IOccupySpace>();
```

Comment:
I don't want to duplicate the code for throwing errors on duplicates here. The lookup literally doesn't matter performance wise - it's only once per actor and that's nothing compared to creating the traits and populating the trait dictionary in the first place. Even if we really did care about the lookup, there's another 10 such lookups just below this loop to worry about as well :)

Simpler is better here.